File: Server/src/main/java/Server/core/game/content/dialogue/GruborDialogue.java
package core.game.content.dialogue;

import core.game.node.entity.npc.NPC;
import core.plugin.Initializable;
import core.game.node.entity.player.Player;

/**
 * Represents the grubor dialogue plugin.
 * @author 'Vexia
 * @version 1.0
 */
@Initializable
public final class GruborDialogue extends DialoguePlugin {

	/**
	 * Constructs a new {@code GruborDialogue} {@code Object}.
	 */
	public GruborDialogue() {
		/**
		 * empty.
		 */
	}

	/**
	 * Constructs a new {@code GruborDialogue} {@code Object}.
	 * @param player the player.
	 */
	public GruborDialogue(Player player) {
		super(player);
	}

	@Override
	public DialoguePlugin newInstance(Player player) {
		return new GruborDialogue(player);
	}

	@Override
	public boolean open(Object... args) {
		npc = (NPC) args[0];
		interpreter.sendDialogues(npc, FacialExpression.HALF_GUILTY, "Yes? What do you want?");
		stage = 0;
		return true;
	}

	@Override
	public boolean handle(int interfaceId, int buttonId) {
		switch (stage) {
		case 0:
			interpreter.sendOptions("Select an Option", "Would you like your hedges trimming?", "I want to come in.", "Do you want to trade?");
			stage = 1;
			break;
		case 1:
			switch (buttonId) {
			case 1:
				interpreter.sendDialogues(player, FacialExpression.HALF_GUILTY, "Would you like your hedges trimming?");
				stage = 10;
				break;
			case 2:
				interpreter.sendDialogues(player, FacialExpression.HALF_GUILTY, "I want to come in.");
				stage = 20;
				break;
			case 3:
				interpreter.sendDialogues(player, FacialExpression.HALF_GUILTY, "Do you want to trade?");
				stage = 30;
				break;
			}
			break;
		case 10:
			interpreter.sendDialogues(npc, FacialExpression.HALF_GUILTY, "Eh? Don't be daft! We don't even HAVE any hehdges!");
			stage = 11;
			break;
		case 11:
			end();
			break;
		case 20:
			interpreter.sendDialogues(npc, FacialExpression.HALF_GUILTY, "No, go away.");
			stage = 21;
			break;
		case 21:
			end();
			break;
		case 30:
			interpreter.sendDialogues(npc, FacialExpression.HALF_GUILTY, "No, I'm busy.");
			stage = 31;
			break;
		case 31:
			end();
			break;
		}
		return true;
	}

	@Override
	public int[] getIds() {
		return new int[] { 789 };
	}
}

File: Server/src/main/content/region/asgarnia/burthorpe/quest/heroesquest/GruborDialogue.kt
package content.region.asgarnia.burthorpe.quest.heroesquest

import content.data.Quests
import core.api.*
import core.game.dialogue.DialogueBuilder
import core.game.dialogue.DialogueBuilderFile
import core.game.dialogue.DialoguePlugin
import core.game.dialogue.FacialExpression
import core.game.node.entity.player.Player
import core.plugin.Initializable
import org.rs09.consts.NPCs

@Initializable
class GruborDialogue (player: Player? = null) : DialoguePlugin(player) {
    override fun newInstance(player: Player): DialoguePlugin {
        return GruborDialogue(player)
    }
    override fun handle(interfaceId: Int, buttonId: Int): Boolean {
        openDialogue(player, GruborDialogueFile(), npc)
        return false
    }
    override fun getIds(): IntArray {
        return intArrayOf(NPCs.GRUBOR_789)
    }
}

class GruborDialogueFile : DialogueBuilderFile() {
    override fun create(b: DialogueBuilder) {

        b.onPredicate { player ->
            getQuestStage(player, Quests.HEROES_QUEST) >= 2 &&
                getAttribute(player, HeroesQuest.attributeGruborLetsYouIn, false) &&
                HeroesQuest.isBlackArm(player)
        }
                .playerl("Hi.")
                .npcl("Hi, I'm a little busy right now.")
                .end()

        b.onPredicate { player ->
            getQuestStage(player, Quests.HEROES_QUEST) >= 2 &&
                !getAttribute(player, HeroesQuest.attributeGruborLetsYouIn, false) &&
                HeroesQuest.isBlackArm(player)
        }
                .npcl(FacialExpression.THINKING, "Yes? What do you want?")
                .options()
                .let { optionBuilder ->

                    optionBuilder.option_playerl("Rabbit's foot.")
                            .npcl("Eh? What are you on about? Go away!")
                            .end()

                    optionBuilder.option_playerl("Four leaved clover.")
                            .npcl("Oh you're one of the gang are you? Ok, hold up a second, I'll just let you in through here.")
                            .linel("You hear the door being unbarred from inside.")
                            .endWith { _, player ->
                                setAttribute(player, HeroesQuest.attributeGruborLetsYouIn, true)
                            }

                    optionBuilder.option_playerl("Lucky horseshoe.")
                            .npcl("Eh? What are you on about? Go away!")
                            .end()

                    optionBuilder.option_playerl("Black cat.")
                            .npcl("Eh? What are you on about? Go away!")
                            .end()
                }


        b.onPredicate { _ -> true }
                .npcl(FacialExpression.THINKING, "Yes? What do you want?")
                .options()
                .let { optionBuilder ->

                    optionBuilder.option_playerl("Would you like your hedges trimming?")
                            .npcl("Eh? Don't be daft! We don't even HAVE any hedges!")
                            .end()

                    optionBuilder.option_playerl("I want to come in.")
                            .npcl("No, go away.")
                            .end()

                    optionBuilder.option_playerl("Do you want to trade?")
                            .npcl("No, I'm busy.")
                            .end()
                }
    }
}